### PR TITLE
Check NetworkManager Version if it is supported

### LINKED ISSUE #2330: Installation in unhealthy state - ConnectivityCheckEnabled
**Systems:**
- Odroid-XU4
- OS: DietPi v6.33.3 (based on Debian 9)
- Supervisor: 2020.12.2 (stable channel)
- Home-Assistant: 0.118.4
- All running on docker 19.03.13
- Connected via ethernet

**Info**

System Health

Home Assistant Core Integration

version: 0.118.4
installation_type: Home Assistant Supervised
dev: false
hassio: true
docker: true
virtualenv: false
python_version: 3.8.6
os_name: Linux
os_version: 4.14.180+
arch: armv7l
timezone: Europe/London

Home Assistant Cloud

logged_in: true
subscription_expiration: December 16, 2020, 12:00 AM
relayer_connected: true
remote_enabled: false
remote_connected: false
alexa_enabled: false
google_enabled: true
can_reach_cert_server: ok
can_reach_cloud_auth: ok
can_reach_cloud: ok

Hass.io

host_os: Debian GNU/Linux 9 (stretch)
update_channel: stable
supervisor_version: 2020.12.2
docker_version: 19.03.13
disk_total: 28.1 GB
disk_used: 14.6 GB
healthy: failed to load: Unhealthy
supported: failed to load: Unsupported
supervisor_api: ok
version_api: ok
installed_addons: Node-RED (7.2.11), Let's Encrypt (4.11.0), Mosquitto broker (5.1), ESPHome (1.15.3), MariaDB (2.2.1), InfluxDB (3.7.9), Samba share (9.3.0), Traccar (0.9.2), Grafana (5.3.6), phpMyAdmin (0.1.4), NGINX Home Assistant SSL proxy (3.0.1), File editor (5.2.0), Zigbee2mqtt (1.16.1), TasmoAdmin (0.13.1)

Lovelace

dashboards: 2
mode: storage
views: 11
resources: 2

**Problem**

Installation was carried out with the convenience script install.sh

Apart from the 'Unsupported OS', everything was running fine.

Today, the Supervisor started throwing out the following error:
"20-12-03 14:24:13 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-374' coro=<HostManager.reload() done, defined at /usr/src/supervisor/supervisor/host/__init__.py:90> exception=KeyError('ConnectivityCheckEnabled')>
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/host/__init__.py", line 99, in reload
    await self.network.update()
  File "/usr/src/supervisor/supervisor/host/network.py", line 107, in update
    await self.check_connectivity()
  File "/usr/src/supervisor/supervisor/host/network.py", line 76, in check_connectivity
    if not self.sys_dbus.network.connectivity_enabled:
  File "/usr/src/supervisor/supervisor/dbus/network/__init__.py", line 58, in connectivity_enabled
    return self.properties[DBUS_ATTR_CONNECTION_ENABLED]
KeyError: 'ConnectivityCheckEnabled'"

I checked that Network-Manager was enabled, re-run the install.sh script without any issues.
I tried to manually set the IP via the Home-Assistant web interface and got a Error 500.

I am not 100% sure where else to look. If I understand the error correctly, the system indicates that it is not connected (to the internet?).

### PR DESCRIPTION
**Pull request by pvizeli**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #2330
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/dbus/network/__init__.py [hunk 1]
```diff
@@ -23,6 +30,8 @@
 
 _LOGGER: logging.Logger = logging.getLogger(__name__)
 
+MINIMAL_VERSION = "1.14.0"
```
Comment (ludeeus) [inline supervisor/dbus/network/__init__.py:33]:
```suggestion
MINIMAL_VERSION = "1.14.6"
```
<https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md#supported-operating-system-system-dependencies-and-versions>

File: supervisor/dbus/network/__init__.py [hunk 2]
```diff
@@ -93,6 +107,28 @@ async def connect(self) -> None:
...
+        """Validate Version of NetworkManager."""
+        self.properties = await self.dbus.get_properties(DBUS_NAME_NM)
+
+        try:
+            if pkg_parse(self.version) >= pkg_parse(MINIMAL_VERSION):
+                return
+        except (TypeError, ValueError, KeyError):
+            pass
+
+        _LOGGER.error("The version of NetworkManager is not supported!")
```
Comment (ludeeus) [inline supervisor/dbus/network/__init__.py:129]:
```suggestion
        _LOGGER.error("Version '%s' of NetworkManager is not supported!", self.version)
```

**Review by ludeeus — COMMENTED** [review]
The version check should also be added to `supervisor/resolution/evaluations/network_manager.py`

**Comment by pvizeli** [comment]
We can't use it in evaluation. We are not connected anymore at this point. Also, DBUS have per design no access to coresys

**Review by ludeeus — APPROVED** [review]

**Pull request merged**